### ISSUE TITLE
False positive in resultMap : properties of parameterized class

### ISSUE THREAD

**Issue by harawata** [harawata/mybatipse#22]
Reported by Mukarram
https://groups.google.com/d/msg/mybatis-user/Re2J3t0oioo/1Uns4NvuzQoJ

```java
public class Animal {
  private String diet;
  public String getDiet() { return diet;}
  public void setDiet(String diet) { this.diet = diet;}
}
```
```java
public class Person extends Animal { 
  private Integer id; 
  private String name; 
} 
```
```xml
<resultMap type="private_result.Person" id="PersonResult"> 
  <id property="id" column="id" /> 
  <result property="name" column="name" /> 
  <result property="diet" column="diet" />  <!-- Error here in Mybatipse -->
</resultMap> 
```

**Comment by harawata** [comment]
Hi @kilokahn ,
I'm sorry for taking so long...I have just released 1.0.10 which contains the fix.
Thank you again for reporting this!
Iwao

**Comment by kilokahn** [comment]
Hello @harawata 

I tried 1.0.10 and it fixes most of the problems - thanks! 

However, there seems to be one other scenario that is not completely correctly handled - I am putting it here but please let me know if I should track it in a separate issue.

Consider this fresh contrived hierarchy:

	public interface Sound {
		String getId();
		void setId(String id);
	}

	public interface Being<S extends Sound> {
		S getSound();
		void setSound(S sound);
	}

	public class BaseBeing<S extends Sound> implements Being<S> {
		private S sound;
		public S getSound() {
			return sound;
		}
		public void setSound(S sound) {
			this.sound = sound;
		}
	}

	public class Animal<S extends Sound> extends BaseBeing<S> {
	}
	public class HumanTalk implements Sound {
		private String id;
		public String getId() {
			return id;
		}
		public void setId(String id) {
			this.id = id;
		}
	}
	public class Person extends Animal<HumanTalk> {
	}

and mapper XML as 

    <resultMap type="com.kilo.domain.Person" id="person">
        <result property="sound.id" column="sound_id"/>
    </resultMap>

Here mybatipse is not able to see that `id` is a valid property on `sound` property of `Person`. 
Interestingly, if we alter the hierarchy to have the `Animal` directly implement the `Being` interface and remove the inheritance on `BaseBeing`, mybatipse validations go through successfully.

Hope the explanation is clear and please feel free to let me know if I can clarify it anymore with a test project. Thanks for the patience so far @harawata !

**Comment by harawata** [comment]
Hi @kilokahn ,

I have verified the issue and am going to fix it in 1.0.11.
Thank you for taking the time for the feedback!

There seems to be no auto-magical way to handle various parameterized type use cases, so I will have to squash them one by one ;-)

**Comment by harawata** [comment]
Hi,

I have just released 1.0.11.
Please let me know if there is any further issue.

Thank you!

**Comment by kilokahn** [comment]
Hello @harawata - this looks great. Thanks for fixes!
